src/main/java/codeu/model/store/basic/UserStore.java

```diff
@@ -121,4 +128,103 @@ public boolean isUserRegistered(String username) {
   public void setUsers(List<User> users) {
     this.users = users;
   }
+
+  /**
+    * Returns the number of users currently recorded
+    */
+  public int getUserCount() {
+    return users.size();
+  }
+
+  /**
+    * Returns the users array
+    */
+  public List<User> getAllUsers(){
+    return users;
+  }
+
+  /**
+    * Returns the user with the most messages sent
+    */
+  public String getTopUser(){
+    getMessageCountForUser();
+    sortUserList();
+    return users.get(0).getName();
+  }
+
+  /**
+    * Iterate through all messages and get the count for the owner of ther messages
+    */
+  public void getMessageCountForUser(){
```

Comment:
Instead of getting message count this way, could you instead keep track by loading it from persistent data when you load the User and create the object and then when a message is sent increment the value in the object?

Reply:
yeah that does sound like a better idea, but then there will be problems with the messages that have already been sent but not yet recorded to the users.messageCount. I will work on it for the live update suggestion :)